diff -up pycups-1.9.47/cupsconnection.c~ pycups-1.9.47/cupsconnection.c
--- pycups-1.9.47/cupsconnection.c~	2014-02-12 12:15:12.360834861 +0000
+++ pycups-1.9.47/cupsconnection.c	2014-02-12 12:29:08.459280169 +0000
@@ -128,7 +128,9 @@ UTF8_from_PyObj (char **const utf8, PyOb
   }
   else if (PyString_Check (obj)) {
     const char *ret;
-    PyObject *unicodeobj = PyUnicode_FromEncodedObject (obj, NULL, NULL);
+    PyObject *unicodeobj = PyUnicode_FromEncodedObject (obj, "utf-8", NULL);
+    if (unicodeobj == NULL)
+      return NULL;
     ret = UTF8_from_PyObj (utf8, unicodeobj);
     Py_DECREF (unicodeobj);
     return ret;
