diff -up system-config-printer-1.4.1/newprinter.py.np-traceback system-config-printer-1.4.1/newprinter.py
--- system-config-printer-1.4.1/newprinter.py.np-traceback	2013-06-07 12:40:22.819125802 +0100
+++ system-config-printer-1.4.1/newprinter.py	2013-06-07 12:40:35.549191403 +0100
@@ -868,7 +868,7 @@ class NewPrinterGUI(GtkGUI):
 
         if self.ppdsloader:
             self.ppdsloader.destroy ()
-            self.ppds_loader = None
+            self.ppdsloader = None
 
         if self.printer_finder:
             self.printer_finder.cancel ()
@@ -3418,7 +3418,7 @@ class NewPrinterGUI(GtkGUI):
 
     def on_tvNPMakes_cursor_changed(self, tvNPMakes):
         path, column = tvNPMakes.get_cursor()
-        if path != None:
+        if path != None and self.ppds != None:
             model = tvNPMakes.get_model ()
             iter = model.get_iter (path)
             self.NPMake = model.get(iter, 1)[0]
