diff -up system-config-printer-1.1.16/glade/NewPrinterWindow.glade.untranslated system-config-printer-1.1.16/glade/NewPrinterWindow.glade
--- system-config-printer-1.1.16/glade/NewPrinterWindow.glade.untranslated	2011-06-14 18:03:38.830340803 +0200
+++ system-config-printer-1.1.16/glade/NewPrinterWindow.glade	2011-06-14 18:11:42.235338629 +0200
@@ -398,7 +398,7 @@
 	  <child>
 	    <widget class="GtkLabel" id="label14">
 	      <property name="visible">True</property>
-	      <property name="label">Name</property>
+	      <property name="label" translatable="yes">Name</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -2629,7 +2629,7 @@ DTR/DSR (Hardware)</property>
 
 					      <child>
 						<widget class="GtkLabel" id="lblNetworkFindNotFound">
-						  <property name="label">No printer was found at that address.</property>
+						  <property name="label" translatable="yes">No printer was found at that address.</property>
 						  <property name="use_underline">False</property>
 						  <property name="use_markup">True</property>
 						  <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -2811,7 +2811,7 @@ DTR/DSR (Hardware)</property>
 	  <child>
 	    <widget class="GtkLabel" id="label15">
 	      <property name="visible">True</property>
-	      <property name="label">Device</property>
+	      <property name="label" translatable="yes">Device</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -3825,7 +3825,7 @@ DTR/DSR (Hardware)</property>
 	  <child>
 	    <widget class="GtkLabel" id="label98">
 	      <property name="visible">True</property>
-	      <property name="label">Class Members</property>
+	      <property name="label" translatable="yes">Class Members</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -3886,7 +3886,7 @@ DTR/DSR (Hardware)</property>
 		  <child>
 		    <widget class="GtkLabel" id="label116">
 		      <property name="visible">True</property>
-		      <property name="label">Try to transfer the current settings</property>
+		      <property name="label" translatable="yes">Try to transfer the current settings</property>
 		      <property name="use_underline">False</property>
 		      <property name="use_markup">False</property>
 		      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -4028,7 +4028,7 @@ DTR/DSR (Hardware)</property>
 	  <child>
 	    <widget class="GtkLabel" id="label114">
 	      <property name="visible">True</property>
-	      <property name="label">Change PPD</property>
+	      <property name="label" translatable="yes">Change PPD</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -4212,7 +4212,7 @@ DTR/DSR (Hardware)</property>
 	  <child>
 	    <widget class="GtkLabel" id="label191">
 	      <property name="visible">True</property>
-	      <property name="label">Installed Options</property>
+	      <property name="label" translatable="yes">Installed Options</property>
 	      <property name="use_underline">False</property>
 	      <property name="use_markup">False</property>
 	      <property name="justify">GTK_JUSTIFY_LEFT</property>
diff -up system-config-printer-1.1.16/glade/PrinterPropertiesDialog.glade.untranslated system-config-printer-1.1.16/glade/PrinterPropertiesDialog.glade
--- system-config-printer-1.1.16/glade/PrinterPropertiesDialog.glade.untranslated	2011-06-14 18:12:36.217342695 +0200
+++ system-config-printer-1.1.16/glade/PrinterPropertiesDialog.glade	2011-06-14 18:12:54.486536430 +0200
@@ -77,7 +77,7 @@
 		      <child>
 			<widget class="GtkLabel" id="label226">
 			  <property name="visible">True</property>
-			  <property name="label">Co_nflicts</property>
+			  <property name="label" translatable="yes">Co_nflicts</property>
 			  <property name="use_underline">True</property>
 			  <property name="use_markup">False</property>
 			  <property name="justify">GTK_JUSTIFY_LEFT</property>
