diff -up system-config-printer-1.4.1/jobviewer.py.statusicon-geometry system-config-printer-1.4.1/jobviewer.py
--- system-config-printer-1.4.1/jobviewer.py.statusicon-geometry	2013-07-02 12:44:38.076105142 +0100
+++ system-config-printer-1.4.1/jobviewer.py	2013-07-02 12:44:54.770626200 +0100
@@ -683,16 +683,19 @@ class JobViewer (GtkGUI):
             if visible:
                 w = self.JobsWindow.get_window()
                 aw = self.JobsAttributesWindow.get_window()
-                (s, area, o) = self.statusicon.get_geometry ()
-                w.set_skip_taskbar_hint (True)
-                if aw != None:
-                    aw.set_skip_taskbar_hint (True)
+                (loc, s, area, o) = self.statusicon.get_geometry ()
 
-                w.property_change ("_NET_WM_ICON_GEOMETRY",
-                                   "CARDINAL", 32,
-                                   Gdk.PROP_MODE_REPLACE,
-                                   list (area))
-                self.JobsWindow.iconify ()
+                if loc:
+                    w.set_skip_taskbar_hint (True)
+                    if aw != None:
+                        aw.set_skip_taskbar_hint (True)
+                    w.property_change ("_NET_WM_ICON_GEOMETRY",
+                                       "CARDINAL", 32,
+                                       Gdk.PROP_MODE_REPLACE,
+                                       list (area))
+                    self.JobsWindow.iconify ()
+                else:
+                    self.JobsWindow.set_visible (False)
             else:
                 self.JobsWindow.present ()
                 self.JobsWindow.set_skip_taskbar_hint (False)
