diff -up system-config-printer-1.1.16/monitor.py.high-cpu-usage system-config-printer-1.1.16/monitor.py
--- system-config-printer-1.1.16/monitor.py.high-cpu-usage	2011-06-14 18:55:31.598527480 +0200
+++ system-config-printer-1.1.16/monitor.py	2011-06-14 18:57:41.852907817 +0200
@@ -363,6 +363,9 @@ class Monitor:
                     return False
 
                 self.watcher.cups_ipp_error (self, e, m)
+                if e == cups.IPP_FORBIDDEN:
+                    return False
+
                 return True
         except RuntimeError:
             cups.setUser (user)
@@ -534,14 +537,16 @@ class Monitor:
 
         try:
             self.sub_id = c.createSubscription ("/", events=events)
+            debugprint ("Created subscription %d" % self.sub_id)
         except cups.IPPError, (e, m):
             self.watcher.cups_ipp_error (self, e, m)
 
         cups.setUser (user)
 
-        self.update_timer = gobject.timeout_add_seconds (MIN_REFRESH_INTERVAL,
-                                                         self.get_notifications)
-        debugprint ("Created subscription %d" % self.sub_id)
+        if self.sub_id != -1:
+            self.update_timer = gobject.timeout_add_seconds (
+                MIN_REFRESH_INTERVAL,
+                self.get_notifications)
 
         if self.monitor_jobs:
             jobs = self.jobs.copy ()
