diff -up system-config-printer-1.4.1/gtkinklevel.py.gdk-color system-config-printer-1.4.1/gtkinklevel.py
--- system-config-printer-1.4.1/gtkinklevel.py.gdk-color	2013-05-09 12:10:43.000000000 +0100
+++ system-config-printer-1.4.1/gtkinklevel.py	2014-08-04 15:55:35.064600683 +0100
@@ -27,7 +27,9 @@ class GtkInkLevel (Gtk.DrawingArea):
         Gtk.DrawingArea.__init__ (self)
         self.connect ('draw', self.draw)
         self._level = level
-        self._color = Gdk.color_parse (color)
+        self._color = None
+        if color:
+            self._color = Gdk.color_parse (color)
         if not self._color:
             self._color = Gdk.color_parse ('#cccccc')
 
