diff -up system-config-printer-1.1.16/probe_printer.py.probe-printer-typo system-config-printer-1.1.16/probe_printer.py
--- system-config-printer-1.1.16/probe_printer.py.probe-printer-typo	2009-12-22 14:44:19.000000000 +0000
+++ system-config-printer-1.1.16/probe_printer.py	2010-07-27 14:20:02.988276298 +0100
@@ -28,6 +28,7 @@ import subprocess
 import threading
 import errno
 import cups
+import gobject
 
 try:
     import pysmb
@@ -193,7 +194,7 @@ class BackgroundSmbAuthContext(pysmb.Aut
 
     def _do_perform_authentication (self):
         result = pysmb.AuthContext.perform_authentication (self)
-        self.do_perform_authentication_result = result
+        self._do_perform_authentication_result = result
         self._gui_event.set ()
         
     def perform_authentication (self):
